Makefile

```diff
@@ -1,73 +1,89 @@
-include .env.dev
-
-ENV_DIR ?= venv
-ENV = $(CURDIR)/$(ENV_DIR)
-CELERY = $(ENV)/bin/celery
-PRECOMMIT = $(ENV)/bin/pre-commit
-PIP = $(ENV)/bin/pip
-PYTHON3 = $(ENV)/bin/python3
-PYTEST = $(ENV)/bin/pytest
-
-DOCKER_FILE ?= docker-compose-developer.yml
-
-define setup_engine_env
-	export `grep -v '^#' .env | xargs -0` && cd engine
+DOCKER_COMPOSE_FILE = docker-compose-developer.yml
+
+# if .env is not present simply touch it
+$(shell touch .env)
+include .env
+
+# if COMPOSE_PROFILES is set in .env use it, otherwise use a default (or what is passed in as an arg)
+ifeq ($(COMPOSE_PROFILES),)
+	COMPOSE_PROFILES=engine,oncall_ui,redis,grafana
+endif
+
+# conditionally assign DB based on what is present in COMPOSE_PROFILES
+ifeq ($(findstring mysql,$(COMPOSE_PROFILES)),mysql)
+	DB=mysql
+else ifeq ($(findstring mysql,$(COMPOSE_PROFILES)),mysql)
+	DB=postgres
+else
+	DB=sqlite
+endif
+
+# conditionally assign BROKER_TYPE based on what is present in COMPOSE_PROFILES
+ifeq ($(findstring redis,$(COMPOSE_PROFILES)),redis)
+	BROKER_TYPE=redis
+else
+	BROKER_TYPE=rabbitmq
+endif
+
+define run_engine_docker_command
+    docker run -it \
+		-v $$(pwd)/engine:/etc/app \
+		-v $$(pwd)/engine/oncall.db:/var/lib/oncall/oncall.db \
+		--env BROKER_TYPE=$(BROKER_TYPE) \
+		--env-file .env.dev \
+		--env-file .env.$(DB).dev \
+		--env-file .env \
+		--network oncall \
+		oncall-oncall_engine \
+		$(1)
 endef
 
-$(ENV):
-	python3.9 -m venv $(ENV_DIR)
-
-bootstrap: $(ENV)
-	$(PIP) install -U pip wheel
-	cp -n .env.dev.example .env.dev
-	cd engine && $(PIP) install -r requirements.txt
-	@touch $@
-
-migrate: bootstrap
-	$(setup_engine_env) && $(PYTHON3) manage.py migrate
-
-clean:
-	rm -rf $(ENV)
-
-lint: bootstrap
-	cd engine && $(PRECOMMIT) run --all-files
-
-dbshell: bootstrap
-	$(setup_engine_env) && $(PYTHON3) manage.py dbshell $(ARGS)
+# touch ./engine/oncall.db if it does not exist and DB is eqaul to sqlite
+# conditionally set BROKER_TYPE env var that is passed to docker-compose, based on COMPOSE_PROFILES make arg
+# start docker-compose
+start:
+ifeq ($(DB),sqlite)
+	@if [ ! -f ./engine/oncall.db ]; then \
+		touch ./engine/oncall.db; \
+	fi
+endif
 
-shell: bootstrap
-	$(setup_engine_env) && $(PYTHON3) manage.py shell $(ARGS)
+	COMPOSE_PROFILES=$(COMPOSE_PROFILES) DB=$(DB) BROKER_TYPE=$(BROKER_TYPE) docker-compose -f $(DOCKER_COMPOSE_FILE) up --remove-orphans -d
 
-test: bootstrap
-	$(setup_engine_env) && $(PYTEST) --ds=settings.dev $(ARGS)
+stop:
```

Comment:
Probably we could also introduce a `purge` command that removes volumes as well?

Reply:
do you mean a `docker system prune` helper command that removes oncall dev resources?

Reply:
Hmm, I was thinking more like adding a `--volumes` flag to the `down` command: https://docs.docker.com/engine/reference/commandline/compose_down/#options

But we could probably do `docker prune` stuff as well